content/guides/getting-started/writing-your-first-component-test.md

```diff
@@ -2,4 +2,1184 @@
+        <button type="submit">Login</button>
+      </fieldset>
+    </form>
+  )
+}
+```
+
+```vue
+<template>
+  <form @submit.prevent="formSubmit" class="login-form">
+    <fieldset>
+      <legend>{{ title }}</legend>
+      <label>
+        Username:
+        <input type="text" name="username" v-model="username" />
+        <span v-if="submitted && !username" class="error">
+          Username is required
+        </span>
+      </label>
+      <label>
+        Password:
+        <input type="password" name="password" v-model="password" />
+        <span v-if="submitted && !password" class="error">
+          Password is required
+        </span>
+      </label>
+      <button type="submit">Login</button>
+    </fieldset>
+  </form>
+</template>
+
+<script>
+import { computed, ref } from 'vue'
+
+export default {
+  name: 'LoginForm',
+  props: {
+    title: String,
+    onLogin: Function,
+  },
+  setup(props) {
+    const title = computed(() => props.title || 'Log In')
+    const username = ref('')
+    const password = ref('')
+    const submitted = ref(false)
+
+    const formSubmit = () => {
+      if (username.value && password.value) {
+        props.onLogin({ username: username.value, password: password.value })
+      }
+      submitted.value = true
+    }
+    return { formSubmit, title, username, password, submitted }
+  },
+}
+</script>
+```
+
+:::
+
+## Adding your first spec file
+
+Assuming you've successfully
+[installed the Cypress app](/guides/getting-started/installing-cypress#Installing)
+and
+[configured your framework](/guides/getting-started/component-framework-configuration),
+now it's time to add your first spec file.
+
+In order to test any component, there needs to be a corresponding spec file.
+Just like the component file is used by the application to render the component,
+the spec file is used by Cypress to test the component.
+
+### Naming the spec file
+
+Components are typically written in a modular fashion, with code, styling and
+spec files existing alongside one another in the same folder.
+
+Spec files should be named similarly to the component file being tested, but
+need to have a slightly different file extension so that Cypress can find and
+load them automatically. By default, this extension is either `.cy.js`,
+`.cy.jsx`, `.cy.ts`, or `.cy.tsx`. For example, for a component file named
+`LoginForm.jsx`, you'd create a corresponding spec file called
+`LoginForm.cy.jsx`.
+
+Note that you can change the
+[`testFiles`](/guides/references/configuration#component) configuration option
+to customize how Cypress looks for spec files.
+
+### Creating the spec file
+
+If you haven't already done so,
+[open the Cypress app](/guides/getting-started/installing-cypress#Opening-Cypress).
+
+Now, create an empty `LoginForm.cy.jsx` spec file next to the `LoginForm.jsx` or
+`LoginForm.vue` component file and open it in your editor. The Cypress app
+should update to show the newly-created file in its list of specs.
```

Comment:
We should probably mention that if they are not using JSX then can just create a file with `.cy.js`